### PR TITLE
refactor: improved provide inject by using a single symbol

### LINKED ISSUE #338: vue3-carousel not working inside modal
hi, I'm trying to implement  vue3-carousel inside modal, it doesn't work until i inspect element or resize page it works. what 's problem ? thank in advance

### LINKED ISSUE #346: Wrong use of aria-hidden
**Describe the bug**
Carousel items are marked as aria-hidden=true when not visible. Aria-hidden should only be applied to non-interactive elements. 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden
The attribute causes WCAG errors when using focusable elements (buttons, inputs, links) in carousel items.

**To Reproduce**

1. Render a carousel with two items, one of them using a focusable element (ie a button)
2. Click on the non-button item
3. Copy page html to https://validator.w3.org/#validate_by_input
4. See error related to 
https://accessibilityinsights.io/info-examples/web/aria-hidden-focus/

**Expected behavior**
No accessibility errors

### LINKED ISSUE #428: Feature Request: Eager Slide Update Before Transition Ends
Currently, the carousel component updates the active slide (`currentSlide`) only after the ongoing transition completes. This behavior can create a delayed user experience, as the slide change does not reflect immediately in scenarios where the `v-model` value is updated programmatically or by user interaction.

#### **Expected Behavior**
The active slide should update eagerly in response to a change in the bound `v-model`, even if a transition is ongoing. This would make the carousel feel more responsive and align with user expectations.

#### **Use Case**
This feature would improve UX in scenarios such as:
- Synchronizing a thumbnail carousel with a main carousel.
- Quickly navigating through slides programmatically.
- Instant feedback for user actions like clicking navigation controls.

#### **Proposed Solution**
- Change the carousel default behavior to be eager on updating `currentSlide` data.

#### **Advantages**
- Enhances responsiveness and interactivity.

### LINKED ISSUE #433: Slides from slots is not working
Using a carousel where the slides are coming from a slot is not working

https://stackblitz.com/edit/vue3-carousel-example-basic-vsybzp?file=src%2FCarousel.vue,src%2FApp.vue,src%2FSlides.vue

This is due to the vNode traversal which is done only once, because slots are not mounted yet at that time. The correct approach is to use provide inject (which is already being used for everything else) to register the slides to the carousel

### PR DESCRIPTION
**Pull request by Tofandel**
fix(reactivity): use provide to register slides instead of traversing nodes (fixes #433, #350)
fix(reactivity): use computed and watchers where possible instead of manual updates
fix(reactivity): update vModel as soon as we are sliding instead of after the transition (fixes #428)
fix(carousel): update sizes during animations (fixes #338)
~~fix(carousel): in vertical mode with height auto, calculate the height of the slides automatically~~
fix(accessibility): make elements inside cloned slides non focusable and increase size of pagination buttons (fixes #346)
fix(accessibility): add arrow keys support when carousel is focused (via tab)
fix(performance): only clone itemsToShow elements instead of all the slides
chore(types): improve typings & add tsc during tests & export properly defined components
chore(package): update scripts in package json with new workflows and make package manager agnostic
fix(package): fix order of browser, require and import
chore(github): add test action for PRs
feat(dev): add development playground

This is quite the big refactor but I couldn't really split this into multiple PRs.
There is no breaking change on the api side.
Feel free to use the new playground to experiment (we could also add it to the docs)

### CODE REVIEW

**Comment by ismail9k** [comment]
Hi @Tofandel,

Thank you so much for this extensive PR and all the effort you’ve put into it! 🎉 I can see how much thought went into these fixes and improvements, and I appreciate the attention to reactivity, and performance.

I’ll review the changes in detail and experiment with the new playground soon. However, I want to flag that some of this work may overlap or conflict with updates I’m currently working on in another branch. I’ll take the necessary steps to reconcile those changes as part of the review process.

Thanks again for contributing so much value. I’ll do my best to handle this in a way that aligns with the project’s goals.

**Comment by Tofandel** [comment]
Thanks, I'm sure there could be some conflicts because it's extensive changes, but I still tried to keep it to a minimum while covering as much as I could

Don't hesitate to ping me on the code if you need some explanations

File: src/components/Carousel.ts [hunk 1]
```diff
@@ -121,37 +137,87 @@ export default defineComponent({
...
+              cloned.style.height = 'auto'
+              track?.append(cloned)
+
+              const height = cloned.getBoundingClientRect().height
+
+              track?.removeChild(cloned)
+              return total + height
+            }, 0) / slides.length
+          slideSize.value = height / multiplierHeight
+        }
```
Comment (ismail9k) [inline src/components/Carousel.ts:212]: This addition introduces significant complexity to the component. For now, let's keep it simple by relying on a fixed height. We can introduce this feature in a separate PR later as this PR is already quite large.

File: src/components/Carousel.ts [hunk 2]
```diff
@@ -62,41 +70,49 @@ export default defineComponent({
...
-
-      return dirMap[dir] || dir
+    const normalizedDir = computed<NormalizedDir>(() => {
+      const dir = config.dir || 'ltr'
+      return dir in DIR_MAP ? DIR_MAP[dir as NonNormalizedDir] : (dir as NormalizedDir)
     })
 
-    const isVertical = computed(() => ['ttb', 'btt'].includes(normalizeDir.value))
+    const indexCbs: Array<(index: number) => void> = []
+    const registerSlide: InjectedCarousel['registerSlide'] = (slide, indexCb) => {
```
Comment (ismail9k) [inline src/components/Carousel.ts:98]: 👍

**Review by ismail9k — CHANGES_REQUESTED** [review]
Thank you for your hard work on this PR! I’ve gone through the changes (although it was quite hard due to the PR size), but I believe your changes are a great addition and in the right direction. I’ve added some feedback.

File: package.json [hunk 3]
```diff
@@ -23,23 +27,27 @@
...
-      "require": "./dist/carousel.min.js"
+      "require": "./dist/carousel.min.js",
+      "import": "./dist/carousel.es.min.js"
     },
     "./dist/*.css": {
-      "import": "./dist/*.css",
-      "require": "./dist/*.css"
+      "require": "./dist/*.css",
+      "import": "./dist/*.css"
+    },
```
Comment (ismail9k) [inline package.json:47]: "./dist/*": "./dist/*"

**Review by ismail9k — APPROVED** [review]

File: playground/App.vue [hunk 4]
```diff
@@ -28,17 +30,17 @@
     <div class="carousel-wrapper">
       <VueCarousel
         v-model="currentSlide"
-        :items-to-show="parseFloat(itemsToShow)"
-        :items-to-scroll="parseFloat(itemsToScroll)"
+        :items-to-show="itemsToShow"
+        :items-to-scroll="itemsToScroll"
         :gap="10"
-        :height="height || 'auto'"
+        :height="parseInt(height) || 'auto'"
```
Comment (Tofandel) [inline playground/App.vue:36]: I added something to handle this correctly on the component side lower (|| parseInt(config.height) == config.height),


This allows for other values like `30vh` or `10%`
though why remove the parseFloat on itemsToShow and toScroll?

File: src/components/Carousel.ts [hunk 5]
```diff
@@ -165,32 +163,13 @@ export default defineComponent({
...
-        const transformArr = parseTransform(el)
-
-        if (transformArr.length == 6) {
-          multiplierWidth *= transformArr[0]
-          multiplierHeight *= transformArr[3]
-        }
-      })
 
       // Calculate size based on orientation
+      const { width, height } = viewport.value.getBoundingClientRect()
```
Comment (Tofandel) [inline src/components/Carousel.ts]: The idea was that if we have a number `height` specified, we don't need to call getBoundingClientRect which is a call that can force a reflow, so more performant

Comment (Tofandel) [reply src/components/Carousel.ts]: The multiplier there was part of the animation check, without it the width inside a transform will not be properly calculated (you can see this in the playground animation)

**Review by Tofandel — COMMENTED** [review]
Forgot to submit my comments, dammit

**Comment by Tofandel** [comment]
I forgot to submit my comment on your review, which is why you didn't see them before you updated the files

I reverted some of them accordingly

File: src/components/Icon.ts [hunk 6]
```diff
@@ -1,43 +1,55 @@
...
-
-Icon.props = { name: String, title: String }
-
-export default Icon
+    title: {
+      type: String,
+    }
+  },
+  setup(props: IconProps) {
+    const carousel = inject(injectCarousel, null)
```
[the remark this answers is not in the document]
Comment (Tofandel) [reply src/components/Icon.ts:32]: I added tests for this component

**Pull request merged**